### PR TITLE
Add CurlHandler.DefaultProxyCredentials

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @davidsh, @cipop, @ericeil, @kapilash, @bartonjs, @joshfree

### CODE REVIEW

File: src/Common/tests/System/Net/TestWebProxies.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Test.Common
+{
+    internal sealed class UseSpecifiedUriWebProxy : IWebProxy
```
Comment (stephentoub) [inline src/Common/tests/System/Net/TestWebProxies.cs]: Nothing changed in these test types; I simply separated them out into their own file.

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs [hunk 2]
```diff
@@ -323,8 +323,14 @@ internal void SetProxyOptions(Uri requestUri)
                 if (_handler.Proxy == null)
                 {
                     // UseProxy was true, but Proxy was null.  Let libcurl do its default handling, 
-                    // which includes checking the http_proxy environment variable.
+                    // which includes checking the http_proxy environment variable.
```
Comment (justinvp) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs]: Nit: trailing whitespace

**Comment by ericeil** [comment]
LGTM

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs]: Thanks.  I'll fix it subsequently rather than kicking off another CI run.

**Pull request merged**